src/theory/symmetry_detect.cpp

```diff
@@ -285,63 +272,31 @@ void SymmetryDetect::processSingletonPartitions(
 
 void SymmetryDetect::collectChildren(Node node, vector<Node>& children)
 {
-  if (theory::quantifiers::TermUtil::isAssoc(node.getKind()))
-  {
-    for(const Node& n : node)
-    {
-      collectChildren(n, node.getKind(), children);
-    }
-  }
-  else
-  {
-    children.insert(children.end(), node.begin(), node.end());
-  }
-}
-
-void SymmetryDetect::collectChildren(Node node, Kind k, vector<Node>& children)
-{
-  if (node.getKind() == k)
-  {
-    for(const Node& n : node)
-    {
-      collectChildren(n, k, children);
-    }
-  }
-  else
-  {
-    children.push_back(node);
-  }
-}
-
-void SymmetryDetect::PartitionTrie::getNewPartition(Partition& part)
-{
-  Trace("sym-dt") << "[sym-dt] Getting the new partitions from the leaves of "
-                     "the partition trie:"
-                  << endl;
-  if (!d_variables.empty())
-  {
-    vector<Node> vars;
-    Node fresh_var =
-        NodeManager::currentNM()->mkSkolem("sym_bk", d_variables[0].getType());
-
-    for (vector<Node>::iterator var_it = d_variables.begin();
-         var_it != d_variables.end();
-         ++var_it)
-    {
-      vars.push_back(*var_it);
-      part.d_var_to_subvar[*var_it] = fresh_var;
-    }
-    part.d_subvar_to_vars[fresh_var] = vars;
-  }
-  else
-  {
-    for (map<Node, PartitionTrie>::iterator part_it = d_children.begin();
-         part_it != d_children.end();
-         ++part_it)
+  Node cur;
+  vector<Node> visit;
+  visit.push_back(node);
+  Kind k = node.getKind();
```

Comment:
You still need to check whether theory::quantifiers::TermUtil::isAssoc holds for k here.